code/datums/components/quality.dm

```diff
@@ -0,0 +1,118 @@
+#define UNUSABLE_QUALITY 0.7
+#define SHODDY_QUALITY 0.8
+#define POOR_QUALITY 0.9
+#define NORMAL_QUALITY 1
+#define GOOD_QUALITY 1.05
+#define FINE_QUALITY 1.1
+#define SUPERIOR_QUALITY 1.2
+#define EXCEPTIONAL_QUALITY 1.3
+#define ARTISAN_QUALITY 1.35
+#define MASTERWORK_QUALITY 1.4
+
+#define ARTIFACT_QUALITY 1.4
+
+
+///This Componenet handles adding quality to items
+/datum/component/quality
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/list/quality_levels = list(UNUSABLE_QUALITY,SHODDY_QUALITY,POOR_QUALITY,NORMAL_QUALITY,GOOD_QUALITY,FINE_QUALITY,SUPERIOR_QUALITY,EXCEPTIONAL_QUALITY,ARTISAN_QUALITY,MASTERWORK_QUALITY,ARTIFACT_QUALITY )
+	///Normal distribution
+	var/quality_level
+
+
+	var/obj/item/parent_item
```

Comment:
Why do you need this var as well as the inherited parent var?

Reply:
parent refers to atom iirc, i need it to be item type